Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3395,8 +3395,22 @@ extension BrowserViewController: PreferencesObserver {
                 
                 // Clear ALL data when going from normal mode to private
                 // The other way around is handled in `removeTab`
-                let clearables: [Clearable] = [CookiesAndCacheClearable()]
-                _ = ClearPrivateDataTableViewController.clearPrivateData(clearables)
+                
+                let alert = UIAlertController(title: "Privacy", message: "Clearing session.. Please Wait..", preferredStyle: .alert)
+                self.presentedViewController?.present(alert, animated: true, completion: nil)
+                
+                //If I don't dispatch after, the WKWebsiteDataStore has no time to sync its process.
+                //This is a bug in WebKit itself. If this isn't done, data is NOT cleared!
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+                    let clearables: [Clearable] = [CookiesAndCacheClearable()]
+                    ClearPrivateDataTableViewController.clearPrivateData(clearables).uponQueue(.main) { [weak self] in
+                        alert.dismiss(animated: true, completion: nil)
+                        
+                        guard let self = self else { return }
+                        self.tabManager.removeAll()
```

Comment:
Do we need to do this again? Its done up at line 3393 just before showing the alert